packages/next/src/server/lib/patch-fetch.ts

```diff
@@ -764,8 +764,11 @@ export function createPatchedFetcher(
           }
 
           if (isCacheableRevalidate && !cachedFetchData) {
+            const shouldUseFetchCache =
+              workStore.isOnDemandRevalidate &&
+              !workStore.isStaleRevalidationRequest
             handleUnlock = await incrementalCache.lock(cacheKey)
-            const entry = workStore.isOnDemandRevalidate
+            const entry = shouldUseFetchCache
```

Comment:
I don't think this condition should change as background revalidates can still use fetch cache unless expired, instead we should be ensuring `workStore.isRevalidate` is true for the `x-next-stale-isr` request and then it handles the same as it would for a background revalidate.